src/window/main_menu.c

```diff
@@ -112,23 +121,39 @@ static void confirm_exit(int accepted, int checked)
 
 static void button_click(int type, int param2)
 {
-    if (type == 1) {
+    switch (type) {
+    case mm_id_new_career: 
         window_new_career_show();
-    } else if (type == 2) {
+    break;
+
+    case mm_id_load_file: 
         window_file_dialog_show(FILE_TYPE_SAVED_GAME, FILE_DIALOG_LOAD);
-    } else if (type == 3) {
+    break;
+
+    case mm_id_city_editor: 
         window_cck_selection_show();
-    } else if (type == 4) {
+    break;
+
+    case mm_id_mission_editor: {
         if (!editor_is_present() || !game_init_editor()) {
             window_plain_message_dialog_show(
                 TR_NO_EDITOR_TITLE, TR_NO_EDITOR_MESSAGE, 1);
         } else {
             sound_music_play_editor();
         }
-    } else if (type == 5) {
+    }
+    break;
+
+    case mm_id_options:
         window_config_show(CONFIG_FIRST_PAGE, 1);
-    } else if (type == 6) {
+    break;
+
+    case mm_id_exit_dlg:
         window_popup_dialog_show(POPUP_DIALOG_QUIT, confirm_exit, 1);
+    break;
+
+    default:
+        assert("unexpected option");
```

Comment:
I know using an assert makes the code "safer", but we don't really use it. I think the game should just silently ignore an invalid option.

Thus, you can also remove the `<assert.h>` include.